lint-tasks.yml

```diff
@@ -0,0 +1,217 @@
+version: "3"
+
+vars:
+  LINT_VENV_DIR: "{{.BUILD_DIR}}/lint-venv"
+  LINT_VENV_CHECKSUM_FILE: "{{.BUILD_DIR}}/lint#venv.md5"
+  # Linter target dirs
+  PYTHON_LINT_DIRS: ["{{.ROOT_DIR}}/clp_ffi_py", "{{.ROOT_DIR}}/tests"]
+  CPP_LINT_DIRS: ["{{.CLP_FFI_PY_SRC_DIR}}"]
+
+tasks:
+  check:
+    cmds:
+      - task: "cmake-check"
+      - task: "cpp-check"
+      - task: "py-check"
+      - task: "yml-check"
+
+  fix:
+    cmds:
+      - task: "cmake-fix"
+      - task: "cpp-fix"
+      - task: "py-fix"
+      - task: "yml-fix"
+
+  cmake-check:
+    deps: ["venv"]
+    cmds:
+      - task: "cmake"
+        vars:
+          FLAGS: "--check"
+
+  cmake-fix:
+    deps: ["venv"]
+    cmds:
+      - task: "cmake"
+        vars:
+          FLAGS: "--in-place"
+
+  cpp-configs:
+    # TODO: remove this once all clang-tidy has been fixed
+    deps: [":config-cmake-project"]
+    cmds:
+      - "tools/yscope-dev-utils/lint-configs/symlink-cpp-lint-configs.sh"
+
+  cpp-check:
+    cmds:
+      - task: "cpp-format-check"
+      # TODO: re-enable this once all clang-tidy has been fixed
+      # - task: "cpp-static-check"
+
+  cpp-fix:
+    cmds:
+      - task: "cpp-format-fix"
+      # TODO: re-enable this once all clang-tidy has been fixed
+      # - task: "cpp-static-fix"
+
+  cpp-format-check:
+    sources: &cpp_format_src_files
+      - "{{.LINT_VENV_CHECKSUM_FILE}}"
+      - "{{.CLP_FFI_PY_SRC_DIR}}/**/*.cpp"
+      - "{{.CLP_FFI_PY_SRC_DIR}}/**/*.h"
+      - "{{.CLP_FFI_PY_SRC_DIR}}/**/*.hpp"
+      - "{{.CLP_FFI_PY_SRC_DIR}}/**/*.inc"
+      - "src/.clang-format"
+      - "{{.TASKFILE}}"
+      - ".clang-format"
```

Comment:
.clang-format is the symlink for tools/yscope-dev-utils/lint-configs/.clang-format
I had a discussion with Kirk on this a while ago. Don't remember the details but I think the conclusion is that it's reasonable to use the symlink instead of directly using the source.

Reply:
Make sense